src/index.ts

```diff
@@ -458,7 +458,7 @@ function generateShShim (src: string, to: string, opts: InternalOptions): string
 
   let sh = `\
 #!/bin/sh
-basedir=$(dirname "$(echo "$0" | sed -e 's,\\\\,/,g')")
+basedir=$(dirname "$(realpath "$0" | sed -e 's,\\\\,/,g')")
```

Comment:
is the `realpath` command available on all supported systems?

can this be fixed in https://github.com/pnpm/pnpm/tree/main/pkg-manager/link-bins instead?